src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.cc

```diff
@@ -0,0 +1,267 @@
+                for (uint32_t counter = 0; counter < numberOfActivePersons; ++counter) {
+                    std::string idstring;
+                    buffer >> idstring;
+                    traciActivePersons.push_back(idstring);
+                }
+
+                // helper method takes care of this
+                processPersonIDList(traciActivePersons);
+
+            } else {
+                // subscription for specific person
+                if (responseVariableID == TraCIConstants::VAR_POSITION) {
+                    ASSERT(variableType == TraCIConstants::POSITION_2D);
+                    buffer >> x;
+                    buffer >> y;
+                } else if (responseVariableID == TraCIConstants::VAR_ROAD_ID) {
+                    ASSERT(variableType == TraCIConstants::TYPE_STRING);
+                    buffer >> edge;
+                } else if (responseVariableID == TraCIConstants::VAR_SPEED) {
+                    ASSERT(variableType == TraCIConstants::TYPE_DOUBLE);
+                    buffer >> speed;
+                } else if (responseVariableID == TraCIConstants::VAR_ANGLE) {
+                    ASSERT(variableType == TraCIConstants::TYPE_DOUBLE);
+                    buffer >> angle;
+                } else if (responseVariableID == TraCIConstants::VAR_TYPE) {
+                    ASSERT(variableType == TraCIConstants::TYPE_STRING);
+                    buffer >> typeID;
+                }else {
+                    throw cRuntimeError("Received unknown person subscription result");
+                }
+            }
+
+        } else {
+            // the status of the variable is not ok
+            ASSERT(variableType == TraCIConstants::TYPE_STRING);
+            std::string errormsg;
+            buffer >> errormsg;
+            if (isSubscribed(responseObjectID)) {
+                if (variableStatus == TraCIConstants::RTYPE_NOTIMPLEMENTED) {
+                    throw cRuntimeError(
+                            "TraCI server reported subscribing to vehicle variable 0x%2x not implemented (\"%s\"). Might need newer version.",
+                            responseVariableID, errormsg.c_str());
+                }
+
+                throw cRuntimeError(
+                        "TraCI server reported error subscribing to vehicle variable 0x%2x (\"%s\").",
+                        responseVariableID, errormsg.c_str());
+            }
+        }
+
+    }
+
+    // make sure we are only entering this section if we got a person that we already subscribed to
+    if (isSubscribed(responseObjectID)) {
+        // we want to deliver an update for this person for the next call to getUpdated()
+        TraCIPerson person(x, y, edge, speed, angle, responseObjectID, typeID);
+        mUpdatedPersons.push_back(person);
+    }
+
+    return idListUpdateReceived;
+}
+
+std::list<TraCIPerson> PersonSubscriptionManager::getUpdated() {
+    std::list<TraCIPerson> temp = mUpdatedPersons;
+    mUpdatedPersons.clear();
```

Comment:
Why does this getter have to modify the state of the Manager? Why not just return a (copied) vector of updated persons? The clearing could be done in the update or explicitly.

Reply:
i changed it so that the clearing has to be done explicitly from outside.

Reply:
Alright.